src/cr/cube/cube_slice.py

```diff
@@ -26,134 +54,88 @@ def _update_args(self, kwargs):
             return kwargs
 
         axis = kwargs.get('axis')
-        if axis is None or isinstance(axis, tuple):
-            # If no axis was passed, we don't need to update anything. If axis
-            # was a tuple, it means that the caller was expecting a 3D cube.
-            return kwargs
+        if isinstance(axis, int):
+            kwargs['axis'] += 1
+
+        hs_dims_key = (
+            'transforms'
+            if 'transforms' in kwargs else
+            'include_transforms_for_dims'
+        )
+        hs_dims = kwargs.get(hs_dims_key)
+        if isinstance(hs_dims, list):
+            kwargs[hs_dims_key] = [dim + 1 for dim in hs_dims]
```

Comment:
This code (and eveything having to do with headers and subtotals) is a mystery to me.  Where can I learn WTF? :)

Reply:
Tried my best to elaborate in comments